terraform-provider-azurerm_v2.96.0_x5.exe plugin crashed

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

Terraform v1.1.6
- Installed hashicorp/azurerm v2.96.0 (signed by HashiCorp)
- Installed hashicorp/azuread v2.18.0 (signed by HashiCorp)

### Affected Resource(s)

datasources (like azurerm_log_analytics_workspace) among others.

### Terraform Configuration Files

Crashes started yesterday with no reason.
(No changes done on the configuration files)

### Debug Output

lots of "Plugin did not respond" like this

│ Error: Plugin did not respond
│ 
│   with module.main.data.azurerm_log_analytics_workspace.log_solution,
│   on ..\datasource_infra.tf line 30, in data "azurerm_log_analytics_workspace" "log_solution":
│   30: data "azurerm_log_analytics_workspace" "log_solution" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.


### Panic Output

Stack trace from the terraform-provider-azurerm_v2.96.0_x5.exe plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x54699d5]

goroutine 730 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/automation.dataSourceAutomationAccountRead(0x68, {0x65ca260, 0xc00135c700})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/automation/automation_account_data_source.go:71 +0x8f5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x6ebcc72, {0x75e3718, 0xc001e78080}, 0xc002f2bf50, {0x65ca260, 0xc00135c700})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:346 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc00054b880, {0x75e3718, 0xc001e78080}, 0xc004b09b80, {0x65ca260, 0xc00135c700})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:569 +0xf7
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc001384888, {0x75e3718, 0xc001e78080}, 0xc009ee1840)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:1133 +0x3a8
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc0001ad780, {0x75e37c0, 0xc002f2b470}, 0xc0039e2500)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:478 +0x37b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x6b184a0, 0xc0001ad780}, {0x75e37c0, 0xc002f2b470}, 0xc00309a5a0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:416 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001fefc0, {0x76338b8, 0xc00033ef00}, 0xc003092a20, 0xc0014e4150, 0xbc47150, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0xc6f
google.golang.org/grpc.(*Server).handleStream(0xc0001fefc0, {0x76338b8, 0xc00033ef00}, 0xc003092a20, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x294

Error: The terraform-provider-azurerm_v2.96.0_x5.exe plugin crashed!

### Expected Behaviour

<!--- What should have happened? --->

### Actual Behaviour

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform plan`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

--- a/internal/services/automation/automation_account_data_source.go
+++ b/internal/services/automation/automation_account_data_source.go
@@ -68,8 +68,10 @@ func dataSourceAutomationAccountRead(d *pluginsdk.ResourceData, meta interface{}
 		}
 		return fmt.Errorf("retreiving Automation Account Registration Information %s: %+v", id, err)
 	}
-	d.Set("primary_key", iresp.Keys.Primary)
-	d.Set("secondary_key", iresp.Keys.Secondary)
+	if iresp.Keys != nil {
+		d.Set("primary_key", iresp.Keys.Primary)
+		d.Set("secondary_key", iresp.Keys.Secondary)
+	}
 	d.Set("endpoint", iresp.Endpoint)
 	return nil
 }

